io/disk/port_bounce.py

```diff
@@ -0,0 +1,314 @@
+
+    def port_bounce(self):
+        '''
+        defins and test for different scenarios
+        '''
+        test_ports = []
+        self.log.info("short port bounce for individual ports:")
+        for port in self.port_ids:
+            test_ports = []
+            test_ports.append(port)
+            for _ in range(self.count):
+                self.porttoggle(test_ports, self.sbt)
+
+        self.log.info("Long port bounce for individual ports:")
+        for port in self.port_ids:
+            test_ports = []
+            test_ports.append(port)
+            for _ in range(self.count):
+                self.porttoggle(test_ports, self.lbt)
+
+        self.log.info("port bounce for all ports:")
+        for _ in range(self.count):
+            self.porttoggle(self.port_ids, 300)
+            time.sleep(20)
+
+    def porttoggle(self, test_ports, sleep_time):
+        '''
+        port bounce starts here
+        '''
+        switch_info = self.fc_run_command("switchshow")
+        self.log.info("Swicth info: %s", switch_info)
+
+        # Port Disable
+        self.log.info("Disable port(s) %s", test_ports)
+        port_input = " ".join(test_ports)
+        try:
+            self.fc_run_command("portdisable %s" % port_input)
+        except CommandFailed as cf:
+            self.log.info("port disable failed for port(s) %s, details: %s",
+                          test_ports, str(cf))
+
+        # Port disable verification both in switch and OS
+        time.sleep(self.verify_sleep_time)
+        self.verify_port_disable(test_ports)
+        self.verify_port_toggle_host(test_ports, "Linkdown")
+        self.mpath_fail_check(test_ports)
+        time.sleep(sleep_time)
+
+        # Port Enable
+        self.log.info("Enable port(s) %s", test_ports)
+        try:
+            self.fc_run_command("portenable %s" % port_input)
+        except CommandFailed as cf:
+            self.log.info("port enable failed for port %s, details: %s",
+                          test_ports, str(cf))
+        time.sleep(self.verify_sleep_time)
+        self.verify_port_enable(test_ports)
+        self.verify_port_toggle_host(test_ports, "Online")
+        time.sleep(self.verify_sleep_time)
+        self.mpath_pass_check(test_ports)
+
+    def verify_port_disable(self, test_ports):
+        '''
+        checking port link status after disabling the switch port
+        '''
+        self.log.info("verifying port disables")
+        switch_info = self.fc_run_command("switchshow")
+        for port in test_ports:
+            self.log.info("verify port %s disable in %s", port, test_ports)
+            port_string = ".*%s.*Disabled" % port
+            obj = re.search(port_string, switch_info)
+            if obj:
+                self.log.info("Port %s is disabled", port)
+            else:
+                self.log.debug("switch_info: %s", switch_info)
+                msg = "Port %s is failed to disable" % port
+                self.failure_list[port] = msg
```

Comment:
`verify_port_disable()` and `verify_port_enable()` are same functions, with just different checks. They can be made into one function with just the check different (similar to `verify_port_toggle_host()`)